### PR TITLE
Implement Rhythm option for arpeggiator

### PR DESCRIPTION
**Pull request by soymonitus**
New menu in arpeggiator: RHYTHM. 
This is inspired by the Rhythm patterns from Novation Launchkey but extended with more patterns that just the two included in the Launchkey (which are 0-0 and 0--0).

This will define which notes play (0) and which notes are rests (-)

NOTE: I would like it to be cherry picked as it is an improvement over the arpeggiator for 1.1

### CODE REVIEW

**Comment by soymonitus** [comment]
@m-m-adams Ready for review

**Comment by soymonitus** [comment]
Several fixes added after my exhaustive testing from last night. I think it is now ready to go

File: src/deluge/modulation/arpeggiator.cpp [hunk 1]
```diff
@@ -439,32 +483,68 @@ void ArpeggiatorForDrum::switchNoteOn(ArpeggiatorSettings* settings, ArpReturnIn
...
-	instruction->arpNoteOn = &arpNote;
+	if (shouldPlayNote) {
+		// Play the note
+		noteCodeCurrentlyOnPostArp = kNoteForDrum + (int32_t)currentOctave * 12;
+		instruction->noteCodeOnPostArp = noteCodeCurrentlyOnPostArp;
+		instruction->arpNoteOn = &arpNote;
+	}
+	else {
+		// Rhythm silence: Don't play the note
+		instruction->noteCodeOffPostArp = ARP_NOTE_NONE;
```
[the remark this answers is not in the document]
Comment (soymonitus) [reply src/deluge/modulation/arpeggiator.cpp:516]: ARP_NOTE_NONE is the default value of the instruction and means "don't do anything at all". Then you have other commands like NOTE_ON and NOTE_OFF. As we want just a silence, we pass ARP_NOTE_NONE

**Comment by soymonitus** [comment]
Applied all feedback. Hopefully it is cleaner now @m-m-adams

**Review by m-m-adams — APPROVED** [review]

**Pull request merged**